modules/@apostrophecms/i18n/index.js

```diff
@@ -66,12 +66,40 @@ module.exports = {
             name: 'forbidden'
           });
         }
-        return self.i18n.init(req, res, next);
-      }
+        return next();
+      },
+      i18nextHttpMiddleware: i18nextHttpMiddleware.handle(self.i18next, {})
     };
   },
   methods(self) {
     return {
+      // Add the i18next resources provided by the specified module,
+      // merging with any existing phrases for the same locales and namespaces
+      addResourcesForModule(module) {
+        if (!module.options.i18n) {
+          return;
+        }
+        const ns = module.options.i18n.ns || 'default';
+        for (const entry of module.__meta.chain) {
+          const localizationsDir = `${entry.dirname}/localizations`;
+          if (!fs.existsSync(localizationsDir)) {
+            continue;
+          }
+          for (const localizationFile of fs.readdirSync(localizationsDir)) {
+            const data = JSON.parse(fs.readFileSync(`${localizationsDir}/${localizationFile}`));
+            const locale = localizationFile.replace('.json', '');
+            self.i18next.addResourceBundle(locale, ns, data, true, true);
+          }
+        }
+      },
+      // Adds i18next resources for modules initialized before the i18n module
+      // itself, called by init. Later modules call addResourcesForModule(self),
+      // making phrases available gradually as Apostrophe starts up
+      addInitialResources() {
```

Comment:
Is this not duplicative? We run `addResourcesForModule` in the modules module as well.

Reply:
See comment in the source. Modules that initialize before the i18n module may contain phrases but can't call it to add them because it does not yet exist. This method accounts for those.